drivers/usb/udc/udc_renesas_ra.c

```diff
@@ -0,0 +1,756 @@
+		return;
+	}
+
+	udc_event_xfer_ctrl_in(dev, buf);
+}
+
+static void udc_event_xfer_ctrl_out(const struct device *dev, struct net_buf *const buf,
+				    uint32_t len)
+{
+	net_buf_add(buf, len);
+
+	udc_ctrl_update_stage(dev, buf);
+
+	if (udc_ctrl_stage_is_status_out(dev)) {
+		udc_ctrl_submit_status(dev, buf);
+	}
+
+	if (udc_ctrl_stage_is_status_in(dev)) {
+		udc_ctrl_submit_s_out_status(dev, buf);
+	}
+}
+
+static void udc_event_xfer_out(const struct device *dev, struct net_buf *const buf, uint32_t len)
+{
+	net_buf_add(buf, len);
+	udc_submit_ep_event(dev, buf, 0);
+}
+
+static void udc_event_xfer_in(const struct device *dev, struct net_buf *const buf)
+{
+	udc_submit_ep_event(dev, buf, 0);
+}
+
+static void udc_event_xfer_complete(const struct device *dev, struct udc_renesas_ra_evt *evt)
+{
+	struct net_buf *buf;
+
+	unsigned int ep = evt->hal_evt.xfer_complete.ep_addr;
+	unsigned int result = evt->hal_evt.xfer_complete.result;
+	unsigned int len = evt->hal_evt.xfer_complete.len;
+
+	buf = udc_buf_get(dev, ep);
+
+	if (buf == NULL) {
+		return;
+	}
+
+	if (result == USBD_XFER_RESULT_SUCCESS) {
+		udc_ep_set_busy(dev, ep, 0);
+		if (ep == USB_CONTROL_EP_IN) {
+			udc_event_xfer_ctrl_in(dev, buf);
+		} else if (ep == USB_CONTROL_EP_OUT) {
+			udc_event_xfer_ctrl_out(dev, buf, len);
+		} else {
+			if (USB_EP_DIR_IS_OUT(ep)) {
+				udc_event_xfer_out(dev, buf, len);
+			} else {
+				udc_event_xfer_in(dev, buf);
+			}
+		}
+	} else {
+		udc_submit_ep_event(dev, buf, -EIO);
+	}
+}
+
+static ALWAYS_INLINE void renesas_ra_thread_handler(void *const arg)
+{
+	const struct device *dev = (const struct device *)arg;
+
+	LOG_DBG("Driver %p thread started", dev);
+	while (true) {
+		struct udc_renesas_ra_evt evt;
+
+		k_msgq_get(&drv_msgq, &evt, K_FOREVER);
+		switch (evt.type) {
+		case UDC_RENESAS_RA_EVT_HAL: {
+			switch (evt.hal_evt.event_id) {
+			case USBD_EVENT_SETUP_RECEIVED:
+				udc_event_xfer_setup(dev, &evt);
+				break;
+
+			case USBD_EVENT_XFER_COMPLETE:
+				udc_event_xfer_complete(dev, &evt);
+				break;
+
+			default:
+				break;
+			}
+			break;
+		}
+
+		case UDC_RENESAS_RA_EVT_XFER:
+			udc_event_xfer_next(dev, evt.ep);
+			break;
+
+		case UDC_RENESAS_RA_EVT_STATUS:
+			udc_event_status_in(dev);
+
+		default:
+			break;
+		}
+	}
+}
+
+static void udc_fake_status_in(const struct device *dev, struct udc_ep_config *const cfg)
+{
+	struct udc_renesas_ra_evt evt = {
+		.type = UDC_RENESAS_RA_EVT_STATUS,
+		.ep = USB_CONTROL_EP_IN,
+	};
+
+	k_msgq_put(&drv_msgq, &evt, K_NO_WAIT);
+}
+
+static int udc_renesas_ra_ep_enqueue(const struct device *dev, struct udc_ep_config *const cfg,
+				     struct net_buf *buf)
+{
+	LOG_DBG("%p enqueue %p", dev, buf);
+
+	udc_buf_put(cfg, buf);
+
+	if (cfg->addr == USB_CONTROL_EP_IN && buf->len == 0) {
+		udc_fake_status_in(dev, cfg);
+		return 0;
+	}
+
+	struct udc_renesas_ra_evt evt = {
+		.type = UDC_RENESAS_RA_EVT_XFER,
+		.ep = cfg->addr,
+	};
```

Comment:
Variable definitions should be at the top of a block. Please fix it everywhere.

Reply:
I fixed it